### DISCUSSION TITLE
Add (selectable) categories for Assets

### DISCUSSION THREAD

**Discussion by RedDevilus** [community/community#7213] [Other Feature Feedback, Questions, & Ideas]
Take for example the Current Situation:
![image](https://user-images.githubusercontent.com/24227051/140093927-e59096cd-dfe0-470a-adcc-49c4d3cc5d06.png)

What if there is an option like multiple checkboxes to filter by platform (OS), 32bit/64bit, debug builds, end-user builds.

Potential way (able to customise as you want to do the categories)
![CategorieAssetsGithub](https://user-images.githubusercontent.com/24227051/140097088-30d50143-94e0-45b0-a56e-8b0e172ef8bc.png)

**Comment by MylesBorins** [comment]
We are definitely interested in improving the experience of identifying platform specific assets and discovering the appropriate asset for the machine that's visiting GitHub. One way we are thinking about doing this is through both naming conventions + configuration in the `release.yml`.

I'll spend a bit thinking about it, perhaps we could use "labels" rather than naming conventions as a way to group artifacts... but I'm not convinced that we would want to use radio buttons to filter the results (or have filtering be an option at all). I much prefer an approach where we sniff the customers user agent and surface the most appropriate results by default (and maybe even have a big download button)

**Reply by RedDevilus** [reply]
Fair enough. I just wanted to make a quick visual aid for how it would look and act but if you could filter it looe maybe pressing an oval button with the text on it to remember it or make a more global option to only show specific platforms or do it on a per-repo base. Just some padding would help a ton for people, esspecialy endusers who have no idea what build fits them the most.

**Reply by francislavoie** [reply]
I'd like to follow up on this.

See https://github.com/caddyserver/caddy/releases/tag/v2.6.0-beta.5, we're experimenting with signing our build artifacts with cosign/sigstore. See https://twitter.com/MohammedSahaf/status/1572022375247663105 for some background. This caused the amount of individual assets we publish to skyrocket.

This makes it really awful for end-users who want to download the actual release binary, because now they have to sift through tons of new files which might not be relevant to them to find the one particular file they want. It might not be obvious to them that the `.tar.gz` (or `.zip` for Windows) is the one they actually want. Looking at the size of the asset does help (because all the rest which are metadata are tiny, compared to the ~14MB actual binary), but it's still overwhelming.

I'm really hoping we get some kind of grouping/categorization functionality for publishes assets, so we can split up the actual binaries in one group, and metadata files in another. That would help a lot for users to find exactly the file they need.
